### ISSUE TITLE
set-column throws IndexOutOfBoundsException if index = column-count

### ISSUE THREAD

**Issue by schaueho** [mikera/vectorz-clj#63]
In vectorz-clj 0.45.0, trying to set a column in column n for an array of size m*n throws an IndexOutOfBoundsException.
Sample code: 

```
user> (matrix/set-current-implementation :vectorz)
:vectorz
user> (-> (matrix/new-array [2 3])
          (matrix/set-column 2 [3 4]))
IndexOutOfBoundsException Matrix with shape [2,3] does not have slice: 2 on dimension 0  mikera.matrixx.impl.ARectangularMatrix.checkRow (ARectangularMatrix.java:93)
```

The exact same code works as expected with clatrix:
```
user> (-> (matrix/new-array [2 3])
          (matrix/set-column 2 [3 4]))
 A 2x3 matrix
 -------------
 0,00e+00  0,00e+00  3,00e+00 
 0,00e+00  0,00e+00  4,00e+00 

```

Setting a column also works without problems for positions 0<n.

**Comment by mikera** [comment]
Confirmed this is a bug. Will create a new release with a fix today.

**Comment by mikera** [comment]
Fixed in 0.46.0

**Comment by schaueho** [comment]
Works now, thanks.
